src/common/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {Component} from 'react';
+
+import CheckboxSVG from 'assets/images/CheckboxIcon';
+
+import styles from './checkbox.scss';
+
+interface IProps {
+	disabled?: boolean;
+	active: boolean;
+	onToggle: (_: boolean) => void;
+}
+
+interface IState {
+	checkboxActive: boolean;
+}
+
+class Checkbox extends Component<IProps, IState> {
+	public static defaultProps = {
+		disabled: false,
+		active: false,
+		onToggle: () => false,
+	};
+
+	public state = {
+		checkboxActive: this.props.active,
+	};
+
+	private readonly checkboxChangeHandler = () => {
+		this.setState({checkboxActive: !this.state.checkboxActive});
+		this.props.onToggle(!this.state.checkboxActive);
```

Comment:
what about a event? if i want block a propagation, or use a preventDefault?

Reply:
It's much trickier than you might think. I think this is more about TypeScript mastering

Reply:
1. in case of toggling current state, use setState as a function. Read about using setState
2. read about event bubbling, stopPropagation and preventDefault
3. block event bubbling in event function
4. change signature of setState function
5. let us know when it's done

Reply:
Fixed